### PR TITLE
feat(react-components, react-drawer, react-message-bar, react-swatch-picker-preview): Add missing context exports for Drawer, MessageBar, and SwatchPicker

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

Some exports were missing which prevent being able to recompose Drawer, MessageBar, and SwatchPicker since useContextValues is required when recomposing.

> This is blocking a partner from bumping their version of react-components.

## New Behavior

Added missing exports.

### CODE REVIEW

**Review by SavannahBourgeois — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">37</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">39</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Avatar-mount_base.html">614</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Avatar-mount.html">645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Button-mount_base.html">299</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Button-mount.html">300</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Field-mount_base.html">1144</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Field-mount.html">1111</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProvider-mount_base.html">699</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProvider-mount.html">712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-mount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-mount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">37</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">39</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/MakeStyles-mount_base.html">847</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/MakeStyles-mount.html">836</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Persona-mount_base.html">1698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/Persona-mount.html">1745</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/SpinButton-mount_base.html">1352</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/SpinButton-mount.html">1385</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/SwatchPicker-mount_base.html">1525</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31328/perf-test-react-components/SwatchPicker-mount.html">1532</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.101 MB`<br />`267.303 kB` | `1.101 MB`<br />`267.321 kB` | `84 B` <br />`18 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.55 kB`<br />`20.584 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `221.825 kB`<br />`62.568 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.037 kB`<br />`14.418 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1eb9ebc1183e1afd59f13896673641ae2b8194d3'>1eb9ebc1183e1afd59f13896673641ae2b8194d3</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 import type { ComponentProps } from '@fluentui/react-utilities';
 import type { ComponentState } from '@fluentui/react-utilities';
+import { ContextSelector } from '@fluentui/react-context-selector';
```
[the remark this answers is not in the document]
Comment (sopranopillow) [reply packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md:11]: that lint rule would be great to have!

[the remark this answers is not in the document]
Comment (sopranopillow) [reply packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md]: Updated swatch picker's imports :)

**Review by ValentinaKozlova — APPROVED** [review]

File: packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hotell) [reply packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md]: https://github.com/microsoft/fluentui/issues/31350

**Pull request merged**